subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/AbstractCompleteFileSystemLocationSnapshot.java

```diff
@@ -47,21 +51,72 @@ public CompleteFileSystemLocationSnapshot store(String absolutePath, int offset,
         return this;
     }
 
+    @Override
+    public FileSystemNode asFileSystemNode(String pathToParent) {
+        return getPathToParent().equals(pathToParent)
+            ? this
+            : new PathCompressingSnapshotWrapper(pathToParent, this);
+    }
+
     @Override
     public FileSystemNode withPathToParent(String newPathToParent) {
         return getPathToParent().equals(newPathToParent)
             ? this
             : new PathCompressingSnapshotWrapper(newPathToParent, this);
     }
 
+    @Override
+    public Optional<MetadataSnapshot> getSnapshot() {
```

Comment:
Should we call this `asSnapshot()` or `toSnapshot()`?

Reply:
I would say no. The `FileSystemNode` has a `MetadataSnapshot`, though it isn't one. I don't think we should be converting the node to a snapshot.